src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -935,11 +872,6 @@ protected final void closeNoLock(String reason) throws ElasticsearchException {
         if (isClosed.compareAndSet(false, true)) {
             assert rwl.isWriteLockedByCurrentThread() || failEngineLock.isHeldByCurrentThread() : "Either the write lock must be held or the engine must be currently be failing itself";
             try {
-                try {
```

Comment:
this is just wrong IMO. The data safety is here the lucene index and the translog. We HAVE TO make sure its synced to disk before we call rollback.